app/controllers/csvs_controller.rb

```diff
@@ -0,0 +1,54 @@
+class CsvsController < ApplicationController
+
+  def index
+    respond_to do |format|
+      @csv_exports = CsvExport.where(user_id: current_user.id)
+      format.html
+      format.json { render json: @csv_exports }
+    end
+  end
+  
+  def show
+    respond_to do |format|
+      @csv_export = CsvExport.find(params[:id])
+      format.html
+      format.json { render json: @csv_export }
+      format.csv { download(params) }
```

Comment:
this is finding the export once outside and once inside the params, can the export be passed to the download method ? ... or can download use @ecs_export ?